Apps/Src/UpdateVelocity.c

```diff
@@ -48,23 +53,56 @@ void Task_UpdateVelocity(void *p_arg)
         uint8_t accelPedalPercent = Pedals_Read(ACCELERATOR);
         uint8_t brakePedalPercent = Pedals_Read(BRAKE);
 
+        //Edge detector to toggle the cruise enable on only the rising edge
+        State currEnable = Switches_Read(CRUZ_EN);
+        if((prevEnable == OFF) && (currEnable == ON)){ //we went from off to on, so we should toggle it
+            prevEnable = ON;
+            cruiseEnable ^= 1; 
+        } else if ((currEnable == OFF)){ //went from on to OFF, so reset the prev state
+            prevEnable = OFF;
+        }
+
+        //Edge detector to toggle the cruise set on only the rising edge
+        State currSet = Switches_Read(CRUZ_ST);
+        if((prevSet == OFF) && (currSet == OFF)){
+            prevSet = ON;
+            cruiseSet ^= 1;
+        } else if ((currSet == OFF)){
+            prevSet = OFF;
+        }
+
```

Comment:
I might be confused but if Cruise enable isn't on, can Cruise set be on? (since you set the speed after the enable and then it will follow that speed once set is enabled).

Reply:
No, you're right. Fixed.